librz/arch/isa_gnu/z80/expressions.c

```diff
@@ -39,723 +39,658 @@
  * ~ + - (unary)  rd_factor
  */
 
-static int do_rd_expr (const char **p, char delimiter, int *valid, int level,
-		       int *check, int print_errors);
+static int do_rd_expr(Z80AssemblerState *state, const char **p, char delimiter, int *valid, int level,
+	int *check, int print_errors);
 
-static int
-rd_number (const char **p, const char **endp, int base)
-{
-  int result = 0, i;
-  char *c, num[] = "0123456789abcdefghijklmnopqrstuvwxyz";
-  if (verbose >= 6)
-    fprintf (stderr, "%5d (0x%04x): Starting to read number of base %d"
-	     "(string=%s).\n", stack[sp].line, addr, base, *p);
-  num[base] = '\0';
-  *p = delspc (*p);
-  while (**p && (c = strchr (num, tolower ((const unsigned char)**p))))
-    {
-      i = c - num;
-      if (verbose >= 7)
-	fprintf (stderr, "%5d (0x%04x): Digit found:%1x.\n", stack[sp].line,
-		 addr, i);
-      result = result * base + i;
-      (*p)++;
-    }
-  if (endp)
-    *endp = *p;
-  *p = delspc (*p);
-  if (verbose >= 7)
-    fprintf (stderr, "%5d (0x%04x): rd_number returned %d (%04x).\n",
-	     stack[sp].line, addr, result, result);
-  return result;
+/* print an error message, including current line and file */
+static void printerr(int error, const char *fmt, ...) {
 }
```

Comment:
is this empty?

Reply:
Yes, in the original code it was also empty but there was code in its body guarded under an `#if 0` preprocessor directive. I saw there was no good reason to keep maintaining that code since it's essentially comments, and I deleted it.

This function in some sense "deserves" deletion anyway because it's not following the same general approach to error reporting followed in the rest of the project, it prints directly to stderr and uses a weird `va_list` type to manipulate variadic arguments. I leave it empty instead of completely removing it because other functions in `expression.c` use it a lot, I plan to replace it with multiple functions or by writing to an error buffer in the state structure in a future PR. I didn't do any of this in this PR because it's already massive as it is.